UI/Marker/EnemySetPlacemark.gd

```diff
@@ -6,26 +6,56 @@ export (PackedScene) var enemy_placemark_packed_scene: PackedScene = preload("re
 export (Resource) var enemy_set: Resource
 
 onready var _enemy_set := enemy_set as EnemySet
-
+onready var set_tod_value = 0
+onready var index_to_delete = []
 
 func _ready() -> void:
 	_enemy_set.connect("changed", self, "_on_enemy_set_changed")
+	SetProvider.connect("selected_day_night", self, "_on_selected_day_night")
+	SelectedListManager.connect("selection_cleared", self, "cleared_delete_list")
 	_on_enemy_set_changed()
+
+func _on_selected_day_night(tod_index: int):
+	set_tod_value = tod_index
 	
 func _on_enemy_set_changed() -> void:
 	# Rebuild children elements
 	for child in $VBoxContainer.get_children():
 		$VBoxContainer.remove_child(child)
-		
-	for index in _enemy_set.get_enemies().size():
-		var enemy: Enemy = _enemy_set.get_enemies()[index]
+	var enemies = _enemy_set.get_enemies()
+
+	for index in enemies.size():
+		var enemy: Enemy = enemies[index]
 		var enemy_placemark: EnemyPlacemark = enemy_placemark_packed_scene.instance()
 		enemy_placemark.enemy = enemy
+		enemy_placemark.connect("placemark_selected", self, "_on_placemark_selected", [index])
+		enemy_placemark.connect("placemark_deselected", self, "_on_placemark_deselected", [index])
 		enemy_placemark.connect("placemark_removed", self, "_on_enemy_removed", [index])
 		$VBoxContainer.add_child(enemy_placemark)
-
+	
+	SetProvider.select_day_night(set_tod_value)
```

Comment:
But why does a single enemy set perform an action that affects all enemy sets only to fix itself? Maybe it could be replaced with a call to _on_selected_day_night?